modules/launcher/src/main/java/org/mule/runtime/module/launcher/log4j2/ArtifactAwareContextSelector.java

```diff
@@ -100,22 +83,60 @@ static ClassLoader resolveLoggerContextClassLoader(ClassLoader classLoader) {
       loggerClassLoader = loggerClassLoader.getParent();
     }
 
-    if (!(loggerClassLoader instanceof ArtifactClassLoader)) {
+    if (loggerClassLoader == null) {
       return SYSTEM_CLASSLOADER;
     } else if (isRegionClassLoaderMember(loggerClassLoader)) {
       loggerClassLoader = loggerClassLoader.getParent();
+
+      if (isPolicyClassLoader(loggerClassLoader) && policyLogsToAppEnabled(loggerClassLoader)) {
```

Comment:
this is expensive to evaluate each time. if this could be moved to the cache it'd be great